src/viam/examples/modules/complex/gizmo/api.cpp

```diff
@@ -0,0 +1,314 @@
+    DoOneResponse response;
+
+    grpc::ClientContext ctx;
+
+    *request.mutable_name() = this->name();
+    request.set_arg1(arg1);
+
+    const grpc::Status status = stub_->DoOne(&ctx, request, &response);
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+
+    return response.ret1();
+}
+
+bool GizmoClient::do_one_client_stream(std::vector<std::string> arg1) {
+    DoOneClientStreamResponse response;
+    grpc::ClientContext ctx;
+
+    auto writer(stub_->DoOneClientStream(&ctx, &response));
+    for (std::string arg : arg1) {
+        DoOneClientStreamRequest curr_req = {};
+        curr_req.set_arg1(arg);
+        if (!writer->Write(curr_req)) {
+            // Stream is broken; stop writing.
+            break;
+        }
+    }
+    writer->WritesDone();
+    grpc::Status status = writer->Finish();
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+
+    return response.ret1();
+}
+
+std::vector<bool> GizmoClient::do_one_server_stream(std::string arg1) {
+    DoOneServerStreamRequest request;
+    grpc::ClientContext ctx;
+
+    auto reader(stub_->DoOneServerStream(&ctx, request));
+    DoOneServerStreamResponse curr_resp = {};
+    std::vector<bool> rets = {};
+    while (reader->Read(&curr_resp)) {
+        rets.push_back(curr_resp.ret1());
+    }
+    grpc::Status status = reader->Finish();
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+
+    return rets;
+}
+
+std::vector<bool> GizmoClient::do_one_bidi_stream(std::vector<std::string> arg1) {
+    grpc::ClientContext ctx;
+
+    auto stream(stub_->DoOneBiDiStream(&ctx));
+    for (std::string arg : arg1) {
+        DoOneBiDiStreamRequest curr_req = {};
+        curr_req.set_arg1(arg);
+        if (!stream->Write(curr_req)) {
+            // Stream is broken; stop writing.
+            break;
+        }
+    }
+    stream->WritesDone();
+
+    DoOneBiDiStreamResponse curr_resp = {};
+    std::vector<bool> rets = {};
+    while (stream->Read(&curr_resp)) {
+        rets.push_back(curr_resp.ret1());
+    }
+
+    grpc::Status status = stream->Finish();
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+
+    return rets;
+}
+
+std::string GizmoClient::do_two(bool arg1) {
+    DoTwoRequest request;
+    DoTwoResponse response;
+
+    grpc::ClientContext ctx;
+
+    *request.mutable_name() = this->name();
+    request.set_arg1(arg1);
+
+    const grpc::Status status = stub_->DoTwo(&ctx, request, &response);
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+
+    return response.ret1();
+}
+
+/* Gizmo registration logic. Automatically registers the Gizmo API upon compiling
+ * this file. */
+
+bool gizmo_init() {
+    Registry::register_resource(Gizmo::static_api(), Gizmo::resource_registration());
+    return true;
+};
+
+// NOLINTNEXTLINE
+const bool gizmo_inited = gizmo_init();
```

Comment:
Can we put all this inside an unnamed namespace, similar to how it's done in other resources?

Reply:
Yep!